src/rtapi/rtapi_compat.c

```diff
@@ -116,27 +116,32 @@ int xenomai_gid()
 int user_in_xenomai_group()
 {
     int numgroups, i;
-    gid_t *grouplist;
+    gid_t *grouplist = NULL;
     int gid = xenomai_gid();
 
     if (gid < 0)
 	return gid;
 
     numgroups = getgroups(0,NULL);
-    grouplist = (gid_t *) calloc( numgroups, sizeof(gid_t));
-    if (grouplist == NULL)
-	return -ENOMEM;
-    if (getgroups( numgroups, grouplist) > 0) {
-	for (i = 0; i < numgroups; i++) {
-	    if (grouplist[i] == (unsigned) gid) {
-		free(grouplist);
-		return 1;
+    if(numgroups > 0) // if there was an error will return -1, if none 0
+	{
+	grouplist = (gid_t *) calloc( numgroups, sizeof(gid_t));
+        if (grouplist == NULL)
+	    return -ENOMEM;
+        if (getgroups( numgroups, grouplist) > 0) {
+	    for (i = 0; i < numgroups; i++) {
+		if (grouplist[i] == (unsigned) gid) {
+		    free(grouplist);
+		    return 1;
+		}
 	    }
+	} else {
+	    free(grouplist);
+	    return errno;
 	}
-    } else {
-	free(grouplist);
-	return errno;
     }
+    if(grouplist != NULL)
```

Comment:
That was the original error coverity reported.

However I then fixed the possibility of getgroups() erroring and returning a -1 and that being used as a value.

In fixing that error checking the value and enclosing the code below in curly brackets, I then made it impossible for the original error to occur :smile: